src/coreclr/vm/frames.cpp

```diff
@@ -37,6 +37,175 @@
 
 #define CHECK_APP_DOMAIN    0
 
+void Frame::GcScanRoots(promote_func *fn, ScanContext* sc)
+{
+#define FRAME_TYPE_NAME(frameType) if (this->GetFrameIdentifier() == FrameIdentifier::frameType) { ((frameType*)this)->GcScanRoots_Impl(fn, sc); return; }
+#include "FrameTypes.h"
+    _ASSERTE(!"Unexpected");
+    return;
```

Comment:
```suggestion
    switch (GetFrameIdentifier())
    {
#define FRAME_TYPE_NAME(frameType) case FrameIdentifier::frameType: ((frameType*)this)->GcScanRoots_Impl(fn, sc); break;
#include "FrameTypes.h"
    default: _ASSERTE(!"Unexpected"); break;
    }
```
Should these rather be switch statements to ensure that the dispatch is efficient? Maybe the compiler is smart enough to convert the battery of if checks into a switch, but I would not count on it.

Reply:
Done